src/entt/entity/handle.hpp

```diff
@@ -47,19 +69,64 @@ struct basic_handle {
     }
 
     /**
-     * @brief Assigns the null object to a handle.
+    * @brief Assigns a registry and an entity to a handle.
+    * @param other Another handle.
+    * @return This handle.
+    */
+    basic_handle & operator=(const basic_handle &other) ENTT_NOEXCEPT {
+        reg = other.reg;
+        entt = other.entt;
+        return *this;
+    }
+
+    /**
+     * @brief Assigns the null object to a handle, invalidating both registry and entity.
      * @return This handle.
      */
     basic_handle & operator=(null_t) ENTT_NOEXCEPT {
-        return (*this = static_cast<entity_type>(null));
+        reg = nullptr;
+        entt = null;
+        return *this;
+    }
+
+    /**
+    * @brief Compares two handles.
+    * @return True if both handles refer to the same registry and the same entity, false otherwise.
+    */
+    bool operator==(const basic_handle &other) const ENTT_NOEXCEPT {
+        return reg == other.reg && entt == other.entt;
+    }
+
+    /**
+    * @brief Compares two handles.
+    * @return False if both handles refer to the same registry and the same entity, true otherwise.
+    */
+    bool operator!=(const basic_handle &other) const ENTT_NOEXCEPT {
+        return !(*this == other);
+    }
+
+    /**
+    * @brief Compares a handle to the null object.
+    * @return True if the handle refers to null registry and null entity, false otherwise.
+    */
+    bool operator==(null_t) const ENTT_NOEXCEPT {
+        return !reg || entt == null;
```

Comment:
How does the registry matter here? I don't think it's important to be honest.
Comparing with the null entity is an obfuscate call to the bool operator at the end of the day. You want to know if the handle is valid.
Probably the right thing would be to remove this operator and make users use `operator bool` for validation.

Reply:
Agreed. Will remove.